docs/sphinx/source/contributing/style_guide.rst

```diff
@@ -107,6 +107,107 @@ Note that Windows users need not have the ``make`` utility installed as pvlib
 includes a ``make.bat`` batch file that emulates its interface.
 
 
+.. _example-docstring:
+
+Example Docstring
+-----------------
+
+Here is a template for a function docstring that tries to encapsulate all
+the features that may be used in any docstring. This is a template, so
+not all sections are required for every function, but is a good starting point.
+
+.. code-block:: python
+
+    def example_function(poa_global, exponents, degree_symbol,
+                        default_arg='someval', optional_arg=None):
+        r"""
+        A brief summary of the function.
+
+        A longer description of the function. This can include references
+        to literature [1]_, websites, and other code elements such as
+        functions (:py:func:`pvlib.location.lookup_altitude`) and
+        classes (:py:class:`pvlib.location.Location`).
+
+        .. versionadded:: 0.0.1
```

Comment:
Can we add a sentence/[link](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#describing-changes-between-versions) here and maybe also namedrop the other options, e.g., versionchanged, deprecated, and versionremoved